### PR TITLE
Add grad kernel of det and slogdet

### PR DESCRIPTION
**Pull request by thisjiang**
### PR types

New features
### PR changes

OPs
### Describe

增加`paddle.linalg.det`的反向代码实现。

参考论文：https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf
**Note：Input 矩阵必须可逆！！！**

# 主要实现
`det_grad`：
`det_grad = (grad * det).unsqueeze(-1).unsqueeze(-2) * x.inverse().transpose(-2, -1)`

`slog_grad`：
`slogdet_grad = grad.unsqueeze(-1).unsqueeze(-2) * x.inverse().conj().transpose(-2, -1)`

这里我们调用相应的C++接口实现

### CODE REVIEW

File: paddle/fluid/operators/determinant_op.h [hunk 1]
```diff
@@ -87,21 +93,97 @@ class DeterminantKernel : public framework::OpKernel<T> {
...
-    auto dout_dim = vectorize(dout->dims());
-    auto* dx =
+    auto* ddet =
         context.Output<framework::Tensor>(framework::GradVarName("Input"));
-    T* dx_data = dx->mutable_data<T>(context.GetPlace());
 
-    int64_t numel = dx->numel();
-    for (int64_t idx = 0; idx < numel; idx++) {
-      dx_data[idx] = static_cast<T>(1);
+    PADDLE_ENFORCE_EQ(grad->dims().size() + 2, input->dims().size(),
```
Comment (zhhsplendid) [inline paddle/fluid/operators/determinant_op.h]: Can `input->dims().size() == 2`, in this case `Out@GRAD` size should be 1 ?

**Review by zhhsplendid — COMMENTED** [review]
Please add some unit test to make sure result is correct

File: paddle/fluid/operators/determinant_op.h (hunk 1 above)
Comment (thisjiang) [reply paddle/fluid/operators/determinant_op.h]: Right, thanks! I'll fix it.

**Comment by thisjiang** [comment]
这个PR由于改动太多废了。。。合并commit到PR https://github.com/PaddlePaddle/Paddle/pull/35785

**Pull request closed without merging**